src/LoggerBase.php

```diff
@@ -16,27 +18,36 @@ abstract class LoggerBase extends \Psr\Log\AbstractLogger
     /**
      * Interpolate context values into the message placeholders.
      * 
-     * @param string|array $message The data to be logged.
+     * @param string $message The data to be logged.
      * @param array $context (Optional:) The array of values to insert into the
      *     corresponding placeholders.
-     * @return mixed The resulting log message.
+     * @return string The resulting log message.
      */
-    protected function interpolate($message, array $context = [])
+    protected function interpolate($message, array $context = []): string
     {
         if (is_string($message)) {
             return $this->interpolateString($message, $context);
         }
         
-        if (is_array($message) && $this->isArraySupported) {
-            return \array_merge($message, $context);
-        }
-        
         return $this->interpolateString(
             var_export($message, true),
             $context
         );
     }
-    
+
+    /**
+     * Interpolate context values into the message placeholders.
+     *
+     * @param array $message The data to be logged.
+     * @param array $context (Optional:) The array of values to insert into the
+     *     corresponding placeholders.
+     * @return array The resulting log message.
+     */
+    protected function interpolateArray($message, array $context = []): array
```

Comment:
When the message is an array, we aren't actually interpolating the given context values into a message string. The array merge is simply returning an array with the combined data, so we could probably just get rid of this `interpolateArray()` method and directly do the `\array_merge()` wherever we were using this method.

Reply:
Done, not pushed yet.